tests/test_connection.py

```diff
@@ -128,6 +128,69 @@ def test_bad_opening(node_factory):
     l2.daemon.wait_for_log('to_self_delay 100 larger than 99')
 
 
+def test_opening_tiny_channel(node_factory):
+    # Test custom min-capacity-sat parameters
+    #
+    #       ----> [l2] (1000) - default
+    #      /
+    #  [l1] ----> [l3] (3000) - less than own initial commit tx
+    #      \
+    #       ----> [l4] (~6000) - enough to cover own initial commit tx
+    #
+    # For each:
+    #  1. Try to establish channel 1sat smaller than min_capacity_sat
+    #  2. Try to establish channel exact min_capacity_sat
+    #
+    # BOLT2
+    # The receiving node MAY fail the channel if:
+    #  - funding_satoshis is too small
+    #  - it considers `feerate_per_kw` too small for timely processing or unreasonably large.
+    #
+    dustlimit = 546
+    reserves = 2 * dustlimit
+    min_commit_tx_fees = 5430
+    min_for_funder = min_commit_tx_fees + dustlimit + 1
+
+    l2_min_capacity = 1000
+    l3_min_capacity = 3000
+    l4_min_capacity = min_for_funder
+
+    l1 = node_factory.get_node()
+    l2 = node_factory.get_node()
+    l3 = node_factory.get_node(options={'min-capacity-sat': l3_min_capacity})
+    l4 = node_factory.get_node(options={'min-capacity-sat': l4_min_capacity})
+
+    l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
+    l1.rpc.connect(l3.info['id'], 'localhost', l3.port)
+    l1.rpc.connect(l4.info['id'], 'localhost', l4.port)
+
+    # Open channel with less than default 1000 sats should be rejected
+    with pytest.raises(RpcError, match=r'channel capacity is 999sat, which is below 1000000msat'):
+        l1.fund_channel(l2, l2_min_capacity + reserves - 1)
+    # Open a channel with exactly the minimal amount for the fundee.
+    # NOTE 1: This will raise an exception at l1, as the funder cannot afford the own fees for the initial_commit_tx.
+    # NOTE 2: m-schmoock thinks the current exception text for this case is misleading :)
+    #         How about the internal: "Funder cannot afford fee on initial commitment transaction"
```

Comment:
+1 to fixing this message.

Reply:
@niftynei what do you think of this 3b78165 solution?

As there are three different abort `return NULL` code paths below `initial_channel_tx` and  `initial_commit_tx`. A good way would be to pass a char** _output pointer so the code below can set the correct reasons to be used in the exception passed to the user.

Now the test will raise this exception: `Could not meet their fees and reserve: Funder cannot afford fee on initial commitment transaction`. Which means the funder is not able to cover the commitment tx fees on their side. If I understand it correct, this is because the own side commit tx fees are actually deducted from the wallet separately?

Note: The non-modifed master code raises this ambigious exception at: https://github.com/ElementsProject/lightning/blob/901bd7c1c30aacaa80e39706fa9a65fd8adf13e3/openingd/openingd.c#L674